File: Java/Examples/BackgroundServiceDemo/app/src/main/java/com/anubys/example/backgroundservicedemo/BaseApplication.java
package com.anubys.example.backgroundservicedemo;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;
import android.util.Log;

/** @Autor Created by Anubys on the 17.02.2020 */


public class BaseApplication extends Application {
    private static final String TAG = BaseApplication.class.getSimpleName();
    public static final String CHANNEL_ID = "exampleServiceChannelID";


    //* ************************************************ *
    //*               L I F E - C Y C L E                *
    //* ************************************************ *
    @Override
    public void onCreate() {
        Log.d(TAG, "TAG - BaseApplication - onCreate()");
        super.onCreate();

        createNotificationChannel();
    }


    //* ************************************************ *
    //*          H E L P E R - M E T H O D E             *
    //* ************************************************ *
    private void createNotificationChannel() {
        Log.d(TAG, "TAG - BaseApplication - createNotificationChannel()");
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel serviceChannel = new NotificationChannel(CHANNEL_ID, getResources().getString(R.string.txt_example_service_channel), NotificationManager.IMPORTANCE_DEFAULT);

            NotificationManager manager = getSystemService(NotificationManager.class);
            if (manager != null) {
                manager.createNotificationChannel(serviceChannel);
            }
        }
    }
}

File: Kotlin/Examples/SharedPreferencesDemo/app/src/main/java/com/anubys/example/sharedpreferencesdemo/BaseApplication.kt
package com.anubys.example.sharedpreferencesdemo

import android.app.Application
import android.util.Log

/** @Author Created by Anubys on the 16.05.2020 */

val sharedPreferences: AppPreference by lazy {
    BaseApplication.appPreference!!
}

class BaseApplication : Application() {
    private val tag = BaseApplication::class.java.simpleName

    companion object {
        var appPreference: AppPreference? = null
    }

    override fun onCreate() {
        Log.v(tag,"TAG - BaseApplication - onCreate()")
        super.onCreate()

        appPreference = AppPreference(this)
    }
}
